stock_scanner_web/controllers/main.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Angel Moya <http://angelmoya.es>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import http
+from openerp.http import request
+
+
+class ScannerWeb(http.Controller):
+
+    @http.route([
+        '/scanner_call/<string:terminal_number>',
+        '/scanner_call/<string:terminal_number>/<string:action>',
+        '/scanner_call/<string:terminal_number>/<string:action>/<string:message>',
+    ], website=True)
+    def scanner_call(self,
+                     terminal_number='',
+                     action='',
+                     message=False,
+                     type='http',
+                     auth='public',
+                     website=True):
+        scanner_hardware = request.env()['scanner.hardware'].sudo()
+        values = {}
+        try:
+            action = int(action)
```

Comment:
I don't understand why you try to case action to integer, as it always should be a string (see in this method : https://github.com/OCA/stock-logistics-barcode/blob/9.0/stock_scanner/models/scanner_hardware.py#L218)